devicetypes/samsungsds/samsung-smart-doorlock.src/samsung-smart-doorlock.groovy

```diff
@@ -0,0 +1,412 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+ 
+ // Update includes : indent issue, configure reporting comments, unnecessary line removed
+ 
+ 
+import physicalgraph.zigbee.zcl.DataType
+
+
+metadata {
+	definition (name: "Samsung Smart Doorlock", namespace: "Samsung SDS", author: "kyun.park", mnmn: "SmartThings", vid: "generic-lock") {
+		capability "Actuator"
+		capability "Lock"
+		capability "Battery"
+		capability "Configuration"
+		capability "Health Check"
+		capability "Refresh"
+
+		fingerprint profileId: "0104", inClusters: "0000, 0001, 0003, 0004, 0005, 0009, 0101", outClusters: "0019", manufacturer: "SAMSUNG SDS"
+        
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"toggle", type:"generic",  decoration:"flat", width:6, height:4) {
+			tileAttribute ("device.lock", key:"PRIMARY_CONTROL") {
+				attributeState "locked", label:'locked', action:"lock.unlock", icon:"st.locks.lock.locked", backgroundColor:"#00A0DC", nextState:"unlocking"
+				attributeState "unlocked", label:'unlocked', icon:"st.locks.lock.unlocked", backgroundColor:"#ffffff"
+				attributeState "unknown", label:"unknown", icon:"st.locks.lock.unknown", backgroundColor:"#ffffff"
+				attributeState "unlocking", label:'unlocking', icon:"st.locks.lock.unlocked", backgroundColor:"#ffffff"
+				attributeState "autolockError", label:'autolock error', icon:"st.locks.lock.unlocked", backgroundColor:"#ff0000"
+			}
+			tileAttribute("device.displayName", key: "SECONDARY_CONTROL") {
+				attributeState "displayName", label: 'Model:  ${currentValue}'
+			}
+		}
+		valueTile("battery", "device.battery", inactiveLabel:false, decoration:"flat", width:6, height:2) {
+			state "battery", label:'${currentValue}% BATTERY', unit:""
+		}
+		main "toggle"
+		details(["toggle", "battery"])
+	}
+}
+
+// Globals - Cluster IDs
+private getCLUSTER_POWER() { 0x0001 }
+private getCLUSTER_DOORLOCK() { 0x0101 }
+private getCLUSTER_ALARM() { 0x0009 }
+
+// Globals - Command IDs
+private getDOORLOCK_CMD_UNLOCK_DOOR() { 0x1F }
+private getDOORLOCK_RESPONSE_OPERATION_EVENT() { 0x20 }
+private getPOWER_ATTR_BATTERY_VOLTAGE() { 0x0020 }
+private getDOORLOCK_ATTR_LOCKSTATE() { 0x0000 }
+private getDOORLOCK_ATTR_DOORSTATE() { 0x0003 }
+private getALARM_ATTR_ALARM_COUNT() { 0x0000 }
+private getALARM_CMD_ALARM() { 0x00 }
+
+/**
+ * Called on app installed
+ */
+def installed() {
+	log.trace "ZigBee DTH - Executing installed() for device ${device.displayName}"
+}
+
+/**
+ * Called on app uninstalled
+ */
+def uninstalled() {
+	def deviceName = device.displayName
+	log.trace "ZigBee DTH - Executing uninstalled() for device $deviceName"
```

Comment:
nit: for consistency with the other log messages you can use `${device.deviceName}` here instead of defining the extra variable

Reply:
this has been changed. to be uploaded.